### ISSUE TITLE
Logging

### ISSUE THREAD

**Issue by ts4s** [jgyates/genmon#24]
Went to check on generator today to see what events it had, however there were no file for this.

To keep from add to the length thread, I start a separate one for this topic. 

The ideal to log event lead to the following (made a path correction to the original post, it was note a few msg later):

"sudo python /home/pi/genmon/genlog.py -o /home/pi/genmon/Filename.csv &

If you are loading it from crontab you need the full path of the log file. It will work without the path if you call the script your self. Also you need the & if you want it running in the background. pkill will kill the background app.

It will only log to the file if you generator changes state (start, stop, exercise, alarm, or service due warning)"

I went looking for the file and can't find it. So I'm not sure what happen here. I stopped and started the start genmon.sh and manually start the generator to cause an event. Refresh my ls and still no file?

**Comment by jgyates** [comment]
I added the & to startgenmon.sh on all loaded modules, including ones that are commented out by default. You are correct that the full path is needed if loaded via crontab. The same thing applies to paths listed in any conf files (full path is always needed if you load via crontab).

ts4s, I could not tell from your post, are you still having issues now that you have the full path listed?

**Comment by ts4s** [comment]
![gendatalog](https://cloud.githubusercontent.com/assets/1710792/26532309/96caaed8-43c2-11e7-82ce-5d5167fa0e5d.jpeg)

here's a photo of my startgenmon.sh

i haven't change anything, yet. This is still my pi3 from the original thread. The pi zero w is still in the works, (needs a case for it and the serial board to keep it clean in my generator).

**Comment by jgyates** [comment]
That looks correct. Let me know if you have any issues. I assume you are saying that you have not connected this to your generator yet so I am guessing you don't have any issue with genlog.py.

**Comment by ts4s** [comment]
no, it's been connected. it's not logging any start, stop, or exercise.

**Comment by jgyates** [comment]
ok. After looking closer at your startgenmon.sh file I am guessing the problem is the -o. It should be -f instead of -o. This was a typo that was fixed from the initial checkin (command line help printed -o but the actual command line parser code looked for -f). Let me know if that does not fix the problem. I just did a fresh checkout of the repository and tested and it is working for me. If it is some other issue I am sure we can track it down.
